llvm/tools/llvm-objdump/llvm-objdump.cpp

```diff
@@ -1264,23 +1264,70 @@ static SymbolInfoTy createDummySymbolInfo(const ObjectFile &Obj,
     return SymbolInfoTy(Addr, Name, Type);
 }
 
-static void
-collectBBAddrMapLabels(const std::unordered_map<uint64_t, BBAddrMap> &AddrToBBAddrMap,
-                       uint64_t SectionAddr, uint64_t Start, uint64_t End,
-                       std::unordered_map<uint64_t, std::vector<std::string>> &Labels) {
+static void collectBBAddrMapLabels(
+    const std::unordered_map<uint64_t, BBAddrMap> &AddrToBBAddrMap,
+    const std::unordered_map<uint64_t, PGOAnalysisMap> &AddrToPGOBBAddrMap,
+    uint64_t SectionAddr, uint64_t Start, uint64_t End,
+    std::unordered_map<
+        uint64_t, std::vector<std::pair<std::string, std::string>>> &Labels) {
   if (AddrToBBAddrMap.empty())
     return;
   Labels.clear();
   uint64_t StartAddress = SectionAddr + Start;
   uint64_t EndAddress = SectionAddr + End;
   auto Iter = AddrToBBAddrMap.find(StartAddress);
+  auto PGOIter = AddrToPGOBBAddrMap.find(StartAddress);
   if (Iter == AddrToBBAddrMap.end())
     return;
-  for (const BBAddrMap::BBEntry &BBEntry : Iter->second.getBBEntries()) {
+  for (size_t I = 0; I < Iter->second.getBBEntries().size(); ++I) {
+    const BBAddrMap::BBEntry &BBEntry = Iter->second.getBBEntries()[I];
     uint64_t BBAddress = BBEntry.Offset + Iter->second.getFunctionAddress();
     if (BBAddress >= EndAddress)
       continue;
-    Labels[BBAddress].push_back(("BB" + Twine(BBEntry.ID)).str());
+
+    std::string LabelString = "BB" + Twine(BBEntry.ID).str();
+    std::string PGOString = "";
+
+    if (!AddrToPGOBBAddrMap.empty()) {
```

Comment:
Agreed. During RFC, we've talked about disabling block frequency on cold functions to save space so there could be future scenarios where some functions do not have PGO data.

Reply:
I've switched it up to a warning. I was making the assumption here that this should be handled in `readBBAddrMap` given the following comment in `ELFTypes.h`:
```c++
 // `PGOAnalyses` will always be the same length as the return value on 
 // success, otherwise it is empty. 
```
which would make this an internal API problem (hence the assert), rather than a user input issue. There is the assumption there that all the sections have PGO data associated with them.

The justification for switching it makes a lot of sense though.

Reply:
Also, just to confirm, when you're referring to moving something up, you're referring to the assertion, right? We would still need to check that the map has a value at that point/isn't empty.

Reply:
If you end up going with the zip_equal, then this assertion/warning should be unreachable. And if you filter when building the AddrToPGOAnalysisMap then the warning would need to be removed.

I think it should be safe to rely on readBBAddrMap after the fix in #77139. If anything, maybe an assertion should be placed in readBBAddrMap before returning on success but that seems better for a separate PR.

Reply:
Switched to `zip_equal` with filtering and removed the warning (as now it would trigger if some functions don't have PGO data) as suggested.

I'll open a separate PR to add an assertion at the end of `readBBAddrMapImpl` and then we should be fine without anything here.